x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/SecurityServerTransportInterceptor.java

```diff
@@ -98,7 +104,9 @@ public AsyncSender interceptSender(AsyncSender sender) {
             @Override
             public <T extends TransportResponse> void sendRequest(Transport.Connection connection, String action, TransportRequest request,
                                                                   TransportRequestOptions options, TransportResponseHandler<T> handler) {
-                if (licenseState.isSecurityEnabled() && licenseState.isAuthAllowed()) {
+                final boolean stateNotRecovered = isStateNotRecovered;
+                final boolean sendWithAuth = (licenseState.isSecurityEnabled() && licenseState.isAuthAllowed()) || stateNotRecovered;
```

Comment:
It seems that if the state is not recovered, then we're going to `sendWithAuth` even if the local node explitily disables security.
If so, then why have the check at all? If it's safe to send the auth context it even if security is disabled, isn't it simpler to just send it for every outgoing request?

Reply:
Same thought as @tvernum here.

Reply:
The interceptor is not registered at all if security is disabled.

https://github.com/elastic/elasticsearch/blob/3b70e943eb08f90aa24e9c34033c8ca2a7211239/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java#L808-L829

This change limits to only sending auth when security *could* possibly be enabled. Once we've recovered the state then the check is based upon the license. By having the check we maintain the ability to assert that all outgoing requests have authentication after the state has been recovered.